Using
```sh
git clone --single-branch --revision 7c37fe92d7aa9c01d7f9ac0fe49492e9b462ff1a https://github.com/NixOS/nixpkgs
cat <<<'NIX' > scipy.nix
let
  nixpkgs = import ./default.nix {
    localSystem = {
      gcc = rec {
        arch = "znver5"; # or "skylake"
        tune = arch;
      };
      system = "x86_64-linux";
    };
  };
  pkgs = nixpkgs.pkgs;
in
pkgs.python3Packages.scipy.overrideAttrs {
  src = pkgs.fetchFromGitHub {
    owner = "scipy";
    repo = "scipy";
    rev = "15f6db88542e2e5f8a8662613458d28b2c02ca5a"; # https://github.com/scipy/scipy/pull/25571/changes/15f6db88542e2e5f8a8662613458d28b2c02ca5a
    hash = "sha256-I35lPGhyIyjrEfa47+iBTrcF6IlBy9MDhaTKSDFxtv8=";
    fetchSubmodules = true;
  };
}
NIX
nix-build --max-jobs 1 scipy.nix
```

with `-m{arch,tune}=skylake`, I'm still failing with `TestSTFT::test_roundtrip_scaling`:
```
=================================== FAILURES ===================================
_______________________ TestSTFT.test_roundtrip_scaling ________________________
[gw4] linux -- Python 3.14.6 /nix/store/wy21hyxq6974rirkbs3qj8fd0p2nfp6z-python3-3.14.6/bin/python3.14

self = <scipy.signal.tests.test_spectral.TestSTFT object at 0x7fffb469ec50>

    def test_roundtrip_scaling(self):
        """Verify behavior of scaling parameter. """
        # Create 1024 sample cosine signal with amplitude 2:
        X = np.zeros(513, dtype=complex)
        X[256] = 1024
        x = np.fft.irfft(X)
        power_x = sum(x**2) / len(x)  # power of signal x is 2
    
        # Calculate magnitude-scaled STFT:
        Zs = stft(x, boundary='even', scaling='spectrum')[2]
    
        # Test round trip:
        x1 = istft(Zs, boundary=True, scaling='spectrum')[1]
        assert_allclose(x1, x, atol=np.finfo(x.dtype).resolution)
    
        # For a Hann-windowed 256 sample length FFT, we expect a peak at
        # frequency 64 (since it is 1/4 the length of X) with a height of 1
        # (half the amplitude). A Hann window of a perfectly centered sine has
        # the magnitude [..., 0, 0, 0.5, 1, 0.5, 0, 0, ...].
        # Note that in this case the 'even' padding works for the beginning
        # but not for the end of the STFT.
        assert_allclose(abs(Zs[63, :-1]), 0.5)
        assert_allclose(abs(Zs[64, :-1]), 1)
        assert_allclose(abs(Zs[65, :-1]), 0.5)
        # All other values should be zero:
        Zs[63:66, :-1] = 0
        # Note since 'rtol' does not have influence here, atol needs to be set:
        assert_allclose(Zs[:, :-1], 0, atol=np.finfo(Zs.dtype).resolution)
    
        # Calculate two-sided psd-scaled STFT:
        #  - using 'even' padding since signal is axis symmetric - this ensures
        #    stationary behavior on the boundaries
        #  - using the two-sided transform allows determining the spectral
        #    power by `sum(abs(Zp[:, k])**2) / len(f)` for the k-th time slot.
        Zp = stft(x, return_onesided=False, boundary='even', scaling='psd')[2]
    
        # Calculate spectral power of Zd by summing over the frequency axis:
        psd_Zp = np.sum(Zp.real**2 + Zp.imag**2, axis=0) / Zp.shape[0]
        # Spectral power of Zp should be equal to the signal's power:
        assert_allclose(psd_Zp, power_x)
    
        # Test round trip:
        x1 = istft(Zp, input_onesided=False, boundary=True, scaling='psd')[1]
>       assert_allclose(x1, x)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0
E       
E       Mismatched elements: 64 / 1024 (6.25%)
E       First 5 mismatches are at indices:
E        [897]: (-5.63313689684079e-22-3.197845218714212e-37j) (ACTUAL), 0.0 (DESIRED)
E        [899]: (-1.1360917766476344e-20-2.5634151331239944e-36j) (ACTUAL), 0.0 (DESIRED)
E        [901]: (-2.961464984263225e-20-1.072378624733862e-35j) (ACTUAL), 0.0 (DESIRED)
E        [903]: (-4.2339380258580164e-20-1.5858773511917491e-35j) (ACTUAL), 0.0 (DESIRED)
E        [905]: (-7.88585911753907e-20+2.57881617913359e-35j) (ACTUAL), 0.0 (DESIRED)
E       Max absolute difference among violations: 4.02863462e-17
E       Max relative difference among violations: inf
E        ACTUAL: array([ 2.000000e+00+0.000000e+00j,  0.000000e+00+0.000000e+00j,
E              -2.000000e+00+2.124592e-17j, ..., -5.006897e-19+5.130432e-34j,
E              -2.000000e+00+3.807177e-17j,  9.526871e-19+3.241935e-33j],
E             shape=(1024,))
E        DESIRED: array([ 2.,  0., -2., ...,  0., -2.,  0.], shape=(1024,))

lib/python3.14/site-packages/scipy/signal/tests/test_spectral.py:1975: AssertionError
=============================== warnings summary ===============================
lib/python3.14/site-packages/scipy/sparse/linalg/_isolve/tests/test_iterative.py::test_maxiter_worsening[gmres-numpy]
lib/python3.14/site-packages/scipy/sparse/linalg/_isolve/tests/test_iterative.py::test_maxiter_worsening[qmr-numpy]
  /nix/store/y2zkqqr5g69zz55zaf2s4w9n67addflz-python3.14-scipy-1.18.0/lib/python3.14/site-packages/scipy/_external/array_api_compat/numpy/_aliases.py:94: ComplexWarning: Casting complex values to real discards the imaginary part
    return np.array(obj, copy=copy, dtype=dtype, **kwargs)

lib/python3.14/site-packages/scipy/sparse/tests/test_64bit.py: 135 warnings
lib/python3.14/site-packages/scipy/sparse/tests/test_base.py: 66 warnings
  /nix/store/nrgp1v1bfimkzvi81n8snf3n7nwcn4yx-python3.14-numpy-2.5.0/lib/python3.14/site-packages/numpy/_core/_methods.py:49: ComplexWarning: Casting complex values to real discards the imaginary part
    return umr_sum(a, axis, dtype, out, keepdims, initial, where)

lib/python3.14/site-packages/scipy/sparse/tests/test_64bit.py: 108 warnings
lib/python3.14/site-packages/scipy/sparse/tests/test_base.py: 54 warnings
  /nix/store/y2zkqqr5g69zz55zaf2s4w9n67addflz-python3.14-scipy-1.18.0/lib/python3.14/site-packages/scipy/sparse/_data.py:73: ComplexWarning: Casting complex values to real discards the imaginary part
    self.data.astype(dtype, casting=casting, copy=True),

lib/python3.14/site-packages/scipy/sparse/tests/test_64bit.py: 27 warnings
lib/python3.14/site-packages/scipy/sparse/tests/test_base.py: 6 warnings
  /nix/store/y2zkqqr5g69zz55zaf2s4w9n67addflz-python3.14-scipy-1.18.0/lib/python3.14/site-packages/scipy/sparse/_dia.py:165: ComplexWarning: Casting complex values to real discards the imaginary part
    self.data.astype(res_dtype), one, row_sums)

lib/python3.14/site-packages/scipy/sparse/tests/test_base.py::TestDOK::test_sum_dtype_pair_of_dtypes
lib/python3.14/site-packages/scipy/sparse/tests/test_base.py::TestDOK::test_sum_dtype_pair_of_dtypes
lib/python3.14/site-packages/scipy/sparse/tests/test_base.py::TestDOK::test_sum_dtype_pair_of_dtypes
lib/python3.14/site-packages/scipy/sparse/tests/test_base.py::TestDOKMatrix::test_sum_dtype_pair_of_dtypes
lib/python3.14/site-packages/scipy/sparse/tests/test_base.py::TestDOKMatrix::test_sum_dtype_pair_of_dtypes
lib/python3.14/site-packages/scipy/sparse/tests/test_base.py::TestDOKMatrix::test_sum_dtype_pair_of_dtypes
  /nix/store/y2zkqqr5g69zz55zaf2s4w9n67addflz-python3.14-scipy-1.18.0/lib/python3.14/site-packages/scipy/sparse/_dok.py:628: ComplexWarning: Casting complex values to real discards the imaginary part
    data = np.array(list(self._dict.values()), dtype=dtype)

lib/python3.14/site-packages/scipy/stats/tests/test_axis_nan_policy.py: 79 warnings
  /nix/store/y2zkqqr5g69zz55zaf2s4w9n67addflz-python3.14-scipy-1.18.0/lib/python3.14/site-packages/scipy/stats/_stats_py.py:11074: SmallSampleWarning: One or more sample arguments is too small; all returned values will be NaN. See documentation for sample size requirements.
    return gmean(x, weights=weights, axis=axis, keepdims=keepdims)

lib/python3.14/site-packages/scipy/stats/tests/test_axis_nan_policy.py: 160 warnings
  /nix/store/nrgp1v1bfimkzvi81n8snf3n7nwcn4yx-python3.14-numpy-2.5.0/lib/python3.14/site-packages/numpy/_core/fromnumeric.py:3862: RuntimeWarning: Mean of empty slice
    return _methods._mean(a, axis=axis, dtype=dtype,

lib/python3.14/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-boxcox_llf-args86-kwds86-1-1-False-<lambda>]
lib/python3.14/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-boxcox_llf-args86-kwds86-1-1-False-<lambda>]
lib/python3.14/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-boxcox_llf-args86-kwds86-1-1-False-<lambda>]
  /nix/store/y2zkqqr5g69zz55zaf2s4w9n67addflz-python3.14-scipy-1.18.0/lib/python3.14/site-packages/scipy/stats/tests/test_axis_nan_policy.py:89: SmallSampleWarning: One or more sample arguments is too small; all returned values will be NaN. See documentation for sample size requirements.
    return stats._morestats._boxcox_llf(data, lmb=lmb, axis=axis)

lib/python3.14/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-yeojohnson_llf-args87-kwds87-1-1-False-<lambda>]
lib/python3.14/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-yeojohnson_llf-args87-kwds87-1-1-False-<lambda>]
lib/python3.14/site-packages/scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-yeojohnson_llf-args87-kwds87-1-1-False-<lambda>]
  /nix/store/y2zkqqr5g69zz55zaf2s4w9n67addflz-python3.14-scipy-1.18.0/lib/python3.14/site-packages/scipy/stats/tests/test_axis_nan_policy.py:95: SmallSampleWarning: One or more sample arguments is too small; all returned values will be NaN. See documentation for sample size requirements.
    return stats._morestats._yeojohnson_llf(data, lmb=lmb, axis=axis)

lib/python3.14/site-packages/scipy/stats/tests/test_qmc.py::TestUtils::test_geometric_discrepancy_mst_with_zero_distances
  /nix/store/y2zkqqr5g69zz55zaf2s4w9n67addflz-python3.14-scipy-1.18.0/lib/python3.14/site-packages/scipy/stats/tests/test_qmc.py:234: UserWarning: Sample contains duplicate points.
    assert_allclose(qmc.geometric_discrepancy(sample, method='mst'), 0.5)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED lib/python3.14/site-packages/scipy/signal/tests/test_spectral.py::TestSTFT::test_roundtrip_scaling - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 64 / 1024 (6.25%)
First 5 mismatches are at indices:
 [897]: (-5.63313689684079e-22-3.197845218714212e-37j) (ACTUAL), 0.0 (DESIRED)
 [899]: (-1.1360917766476344e-20-2.5634151331239944e-36j) (ACTUAL), 0.0 (DESIRED)
 [901]: (-2.961464984263225e-20-1.072378624733862e-35j) (ACTUAL), 0.0 (DESIRED)
 [903]: (-4.2339380258580164e-20-1.5858773511917491e-35j) (ACTUAL), 0.0 (DESIRED)
 [905]: (-7.88585911753907e-20+2.57881617913359e-35j) (ACTUAL), 0.0 (DESIRED)
Max absolute difference among violations: 4.02863462e-17
Max relative difference among violations: inf
 ACTUAL: array([ 2.000000e+00+0.000000e+00j,  0.000000e+00+0.000000e+00j,
       -2.000000e+00+2.124592e-17j, ..., -5.006897e-19+5.130432e-34j,
       -2.000000e+00+3.807177e-17j,  9.526871e-19+3.241935e-33j],
      shape=(1024,))
 DESIRED: array([ 2.,  0., -2., ...,  0., -2.,  0.], shape=(1024,))
= 1 failed, 87957 passed, 8364 skipped, 299 xfailed, 22 xpassed, 650 warnings in 265.78s (0:04:25) =
```

--- a/scipy/signal/tests/_scipy_spectral_test_shim.py
+++ b/scipy/signal/tests/_scipy_spectral_test_shim.py
@@ -17,7 +17,6 @@ as the existing one and delivers comparable results. Furthermore, the
 wrappers highlight the different philosophies of the implementations,
 especially in the border handling.
 """
-import platform
 from typing import cast, Literal
 
 import numpy as np
@@ -296,10 +295,11 @@ def istft_compare(Zxx, fs=1.0, window='hann', nperseg=None, noverlap=None,
     atol = np.finfo(x.dtype).resolution*2  # instead of default atol = 0
     rtol = 1e-7  # default for np.allclose()
 
-    # Relax atol on 32-Bit platforms a bit to pass CI tests.
-    #  - Not clear why there are discrepancies (in the FFT maybe?)
-    #  - Not sure what changed on 'i686' since earlier on those test passed
-    if x.dtype == np.float32 and platform.machine() == 'i686':
+    # Relax tolerances for float32, which loses precision in the FFT
+    # roundtrip. Originally observed on i686; also seen on aarch64 and on
+    # builds using FMA contraction (gh-25488), so the drift is dtype-driven
+    # rather than platform-specific.
+    if x.dtype == np.float32:
         # float32 gets only used by TestSTFT.test_roundtrip_float32() so
         # we are using the tolerances from there to circumvent CI problems
         atol, rtol = 1e-4, 1e-5

--- a/scipy/signal/tests/test_fir_filter_design.py
+++ b/scipy/signal/tests/test_fir_filter_design.py
@@ -770,7 +770,7 @@ class TestMinimumPhase:
         H_mag = xp.abs(rfft(h, N))
         H_min_mag = xp.abs(rfft(h_min_sig, N))
         error = H_mag - H_min_mag
-        atol = dict(float32=1e-5, float64=1e-13)[dtype]
+        atol = dict(float32=2e-5, float64=1e-13)[dtype]
         xp_assert_close(error, xp.zeros_like(error), atol=atol)
 
 

--- a/scipy/signal/tests/test_spectral.py
+++ b/scipy/signal/tests/test_spectral.py
@@ -1942,7 +1942,7 @@ class TestSTFT:
 
         # Test round trip:
         x1 = istft(Zs, boundary=True, scaling='spectrum')[1]
-        assert_allclose(x1, x)
+        assert_allclose(x1, x, atol=np.finfo(x.dtype).resolution)
 
         # For a Hann-windowed 256 sample length FFT, we expect a peak at
         # frequency 64 (since it is 1/4 the length of X) with a height of 1
@@ -1972,7 +1972,7 @@ class TestSTFT:
 
         # Test round trip:
         x1 = istft(Zp, input_onesided=False, boundary=True, scaling='psd')[1]
-        assert_allclose(x1, x)
+        assert_allclose(x1, x, atol=np.finfo(x.dtype).resolution)
 
         # The power of the one-sided psd-scaled STFT can be determined
         # analogously (note that the two sides are not of equal shape):
@@ -1992,7 +1992,7 @@ class TestSTFT:
 
         # Test round trip:
         x1 = istft(Zp0, input_onesided=True, boundary=True, scaling='psd')[1]
-        assert_allclose(x1, x)
+        assert_allclose(x1, x, atol=np.finfo(x.dtype).resolution)
 
 
 class TestSampledSpectralRepresentations:

